oneops-admin/lib/shared/exec-order.rb

```diff
@@ -26,6 +26,67 @@
 json_context      = ARGV[1]
 cookbook_path     = ARGV[2] || ''
 service_cookbooks = ARGV[3] || ''
+
+#
+# Check to see if we have older Ruby version
+# We are not modifying anything. Let's the Ruby
+# system kicks off this, if we detects older
+# version when can kick off the update process
+# only if it's hasn't been done.  If the update
+# has been performed then it will use the newer
+# Ruby instead of system Ruby.
+#
+if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new("2.0.0")
+  valid_url_cmd = "curl --connect-timeout 10 -X HEAD #{ENV['RUBY2_BINARY_proxy']}"
+  install_ruby_cmd = "curl #{ENV['RUBY2_BINARY_proxy']} | tar Pxz  && /home/oneops/ruby/ruby-2.0.0-p648/bin/gem install bundler --no-ri --no-rdoc  && chown -R oneops:oneops /home/oneops/ruby"
```

Comment:
Can't we detect the `ruby_binbary_path` depending on the version installed from `RUBY2_BINARY_proxy`? Hard cording to a specific minor version (`2.0.0-p648`) of ruby seems very fragile to me. Things can break if you change `RUBY2_BINARY_proxy` value in future.